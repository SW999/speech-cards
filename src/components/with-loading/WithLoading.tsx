import React, { Suspense } from 'react';
import Loading from '../loading/Loading';

const WithLoading = ({ component: Component, ...rest }) => (
  <Suspense fallback={<Loading />}>
    <Component {...rest} />
  </Suspense>
);

export default WithLoading;
