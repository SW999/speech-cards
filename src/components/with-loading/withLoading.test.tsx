import React from 'react';
import { render } from '@testing-library/react';
import WithLoading from './WithLoading';

const TestComponent = () => <div>Test</div>;

describe('WithLoading component', () => {
  it('WithLoading renders correctly', () => {
    const { asFragment } = render(<WithLoading component={TestComponent} />);

    expect(asFragment()).toMatchSnapshot('WithLoading');
  });
});
