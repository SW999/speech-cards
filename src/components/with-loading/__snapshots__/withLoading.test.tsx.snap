// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoading component WithLoading renders correctly: WithLoading 1`] = `
<DocumentFragment>
  <div>
    Test
  </div>
</DocumentFragment>
`;
